scriptworker/github.py

```diff
@@ -12,6 +13,8 @@
 )
 
 _GIT_FULL_HASH_PATTERN = re.compile(r'^[0-9a-f]{40}$')
+_branch_commits_cache = {}
```

Comment:
I figured out why Chain of Trust didn't contact github.com anymore. This variable lives as long as the program runs. Some parts of beetmover_scriptworker are killed after the worker stopped running the task, but not this very one. Therefore, the first time CoT encounters the URL, it does cache the response, but then it keeps it for waaaaay to long. I think we need to cache it differently. 

I know it must be done at the class the level, because the 4 requests we see are spawned by https://github.com/mozilla-releng/scriptworker/blob/646d478420a213544aa1954c24c53893dc1c4f67/scriptworker/task.py#L441

Maybe the easier way is to store the result in scriptworker's work_dir. If I remember correctly, this one gets nuked after a task is processed.